src/core/integrator_wrappers.jl

```diff
@@ -0,0 +1,195 @@
+export stroboscopicmap, projectedsystem
+
+
+"""
+	stroboscopicmap(ds::ContinuousDynamicalSystem, T; kwargs...)  → smap
+
+Return a map (integrator) that produces iterations over a period T of the `ds`.
+
+You can progress the map one step on the section by calling `step!(smap)`,
+which also returns the next state. You can also set the integrator to start from a new
+state `u` by using `reinit!(smap, u)` and then calling `step!` as normally.
+
+## Keyword Arguments
+* `u0`: initial state
+* `diffeq` is a `NamedTuple` (or `Dict`) of keyword arguments propagated into
+  `init` of DifferentialEquations.jl.
+
+
+## Example
+```julia
+F = 0.27; ω = 0.1;  # smooth boundary
+ds = Systems.duffing(zeros(2), ω = ω, f = F, d = 0.15, β = -1)
+smap = stroboscopicmap(ds, 2*pi/ω; diffeq = (;reltol = 1e-8))
+reinit!(smap,[1., 1.])
+step!(smap)
+u = get_state(smap)
+```
+"""
+function stroboscopicmap(ds::CDS{IIP, S, D}, T = nothing; u0 = get_state(ds),
+	diffeq = NamedTuple(), kwargs...
+	) where {IIP, S, D}
+
+    if !isempty(kwargs)
+        @warn DIFFEQ_DEP_WARN
+        diffeq = NamedTuple(kwargs)
+    end
+
+	if isnothing(T)
+		@warn "T must be defined, taking T=1 as default"
+		T =1
+	end
+
+	integ = integrator(ds, u0; diffeq)
+	return StroboscopicMap(integ, T)
+end
+
+mutable struct StroboscopicMap{I, F}
+	integ::I
+	T::F
+end
+
+function DynamicalSystemsBase.step!(smap::StroboscopicMap)
+	step!(smap.integ, smap.T, true)
+	return smap.integ.u
+end
+function DynamicalSystemsBase.reinit!(smap::StroboscopicMap, u0)
+	reinit!(smap.integ, u0)
+	return
+end
+function DynamicalSystemsBase.get_state(smap::StroboscopicMap)
+	return smap.integ.u
+end
+
+function Base.show(io::IO, smap::StroboscopicMap)
+    println(io, "Iterator of the Stroboscopic map")
+    println(io,  rpad(" rule f: ", 14),     DynamicalSystemsBase.eomstring(smap.integ.f.f))
+    println(io,  rpad(" Period: ", 14),     smap.T)
+end
+
+
+
+"""
+	projectedsystem(ds::ContinuousDynamicalSystem, Δt; kwargs...)  → psys
```

Comment:
Just an idea, but can we maybe generalize this to arbitrary projections? i.e., a user defines a function that given the state of the system returns another lower dimensional one?

We could do this via a keyword `projection = 1:dimension(ds)`. If `projection isa AbstractVector{Int}`, the code is as now. But if `projection isa Function` then we go the new route and get the projected state by just applying `projection(get_state(integ))`.